### PR TITLE
Add damageable objects support

### PR DESCRIPTION
**Pull request by TheNormalnij**
Damageable models can be loaded via a IMG, but they cause random crashes.
This PR adds proper support for damageable models. 

```lua
local modelId = engineRequestModel('object-damageable')
```

### CODE REVIEW

**Comment by Fernando-A-Rocha** [comment]
Do we really need another type "object-damageable"?

File: Client/game_sa/CModelInfoSA.cpp [hunk 1]
```diff
@@ -2064,6 +2089,12 @@ bool CModelInfoSA::IsTowableBy(CModelInfo* towingModel)
     return isTowable;
 }
 
+bool CModelInfoSA::IsDamageableAtomic()
+{
+    void* asDamagable = ((void* (*)())m_pInterface->VFTBL->AsDamageAtomicModelInfoPtr)();
```
Comment (TracerDS) [inline Client/game_sa/CModelInfoSA.cpp:2111]: Maybe just `((void*)...)() != nullptr`?

File: Client/game_sa/CModelInfoSA.cpp [hunk 2]
```diff
@@ -1744,6 +1744,24 @@ void CModelInfoSA::MakeObjectModel(ushort usBaseID)
...
 
+void CModelInfoSA::MakeObjectDamageableModel(std::uint16_t baseModel)
+{
+    CDamageableModelInfoSAInterface* m_pInterface = new CDamageableModelInfoSAInterface();
+
+    CDamageableModelInfoSAInterface* pBaseObjectInfo = static_cast<CDamageableModelInfoSAInterface*>(ppModelInfo[baseModel]);
+    MemCpyFast(m_pInterface, pBaseObjectInfo, sizeof(CDamageableModelInfoSAInterface));
+    m_pInterface->usNumberOfRefs = 0;
+    m_pInterface->pRwObject = nullptr;
+    m_pInterface->usUnknown = 65535;
```
Comment (FileEX) [inline Client/game_sa/CModelInfoSA.cpp:1772]: Set value to -1 for "usUnknown"
https://github.com/multitheftauto/mtasa-blue/blob/ee4593b1a89d36f2a907776323d695f00bdf3d09/Client/game_sa/CModelInfoSA.cpp#L1788

**Comment by TheNormalnij** [comment]
> Do we really need another type "object-damageable"?

This is last missing type from the game world, that can not be allocated. SA uses IS_DAMAGEABLE model flag to allocate damageable models. Changing this flag in MTA causes crashes.  So we need to allocate those models explicitly

**Review by botder — DISMISSED** [review]

**Pull request merged**